client/container_create.go

```diff
@@ -103,14 +105,16 @@ func (cli *Client) ContainerCreate(ctx context.Context, config *container.Config
 	return response, err
 }
 
-// formatPlatform returns a formatted string representing platform (e.g. linux/arm/v7).
+// formatPlatform returns a formatted string representing platform (e.g., "linux/arm/v7").
+//
+// It is a fork of [platforms.Format], and does not yet support "os.version",
+// as [[platforms.FormatAll] does.
```

Comment:
I'll be opening a ticket for this; the TL;DR is that `docker container create` does not currently support passing `os.version`, which gets silently discarded, but the daemon _would_ be able to handle it.


I'm considering that perhaps we should move away from using the `?platform=` query-arg (which uses the string format, not JSON), and adding information in the body (HostConfig?) that can hold information that was passed to resolve the image (name of image, requested platform?)